lib/board.ex

```diff
@@ -24,46 +24,51 @@ defmodule Board do
 
   @spec move_status(map(), integer()) :: atom()
   def move_status(board, player_move) do
-    if is_out_of_bounds?(board, player_move) do
-      :out_of_bounds
-    else
-      :valid
+    cond do
+      is_out_of_bounds?(board, player_move) -> :out_of_bounds
+      is_space_taken?(board, player_move) -> :space_taken
+      true -> :valid
     end
   end
 
+  @spec is_space_taken?(map(), integer()) :: boolean()
+  defp is_space_taken?(board, player_move) do
+    Map.fetch!(board, player_move) != :empty
+  end
+
   @spec is_out_of_bounds?(map(), integer()) :: boolean()
   defp is_out_of_bounds?(board, player_move) do
     player_move not in 1..Enum.count(board)
   end
 
   @spec has_player_won?(map()) :: boolean()
   defp has_player_won?(board) do
-    Enum.any?([get_all_rows_status(board), get_all_columns_status(board), get_both_diagonals_status(board)])
+    Enum.any?([all_rows_status(board), all_columns_status(board), both_diagonals_status(board)])
   end
 
   @spec is_board_full?(map) :: boolean
   defp is_board_full?(board) do
     !Enum.member?(Map.values(board), :empty)
   end
 
-  @spec get_board_dimensions(map()) :: integer()
-  def get_board_dimensions(board) do
+  @spec board_dimensions(map()) :: integer()
+  def board_dimensions(board) do
     trunc(:math.sqrt(Enum.count(board)))
   end
 
-  @spec get_all_rows_status(map()) :: boolean()
-  defp get_all_rows_status(board) do
-    board_dimensions = get_board_dimensions(board)
+  @spec all_rows_status(map()) :: boolean()
+  defp all_rows_status(board) do
+    board_dimensions = board_dimensions(board)
     board
     |> Map.values()
     |> Enum.chunk_every(board_dimensions)
     |> Enum.map(&Enum.uniq/1)
     |> Enum.any?(&is_winning_vector?/1)
   end
 
-  @spec get_all_columns_status(map()) :: boolean()
-  defp get_all_columns_status(board) do
-    board_dimensions = get_board_dimensions(board)
+  @spec all_columns_status(map()) :: boolean()
```

Comment:
Same as above. Maybe we can call this `won_by_column?`?

Reply:
Same as above, I think `win_by_column` is a much better name! Changing it now, thank you!